src/App.tsx

```diff
@@ -1,16 +1,24 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [preparedMovies, setPreparedMovies] = useState(moviesFromServer);
+
+  const addPreparedMovie = (movie: Movie) => {
+    setPreparedMovies([...preparedMovies, movie]);
```

Comment:
In order to work with the actual state, you need to take it from the callback:
```suggestion
    setPreparedMovies((prevMovieList) => [...prevMovieList, movie]);
```